specifyweb/frontend/js_src/lib/components/HomePage/index.tsx

```diff
@@ -20,13 +25,22 @@ const taxonTiles = f.store(() => (
 
 export function WelcomeView(): JSX.Element {
   const [mode] = userPreferences.use('welcomePage', 'general', 'mode');
+  const [query = ''] = useSearchParameter('q');
+  const value = React.useState(query);
+  const [pendingQuery] = value;
```

Comment:
why not move these into the `<SearchForm />`?

`formId` is the only one that is needed outside of it